RecoDataProducts/src/TrkToCaloExtrapol.cc

```diff
@@ -129,26 +129,13 @@ namespace mu2e {
 
 
   Hep3Vector TrkToCaloExtrapol::t0Momentum() const{
```

Comment:
This function name implies some relation to t0, but the implementation has nothing to do with t0.   Getting the value at t0 (ie z=0) is trivial as the helix knows the flightlength at 0.  It seems odd too that extrapolating to the calorimeter would want to use the momentum at the 1st hit, which is the farthest away measurement. I suggest re-thinking the calling code and re-implementing the function as needed.

Reply:
I propose to:
- rename them "exitMomentum" and "exitMomentumErr"
- use as pathLength the data member _pathLengthExit
in this way, these functions will be consistent with the other functions named "exit<something>".

Would that work?

Reply:
The TrkToCaloExtrapol object is initialized with the pathLength at which the track enters in the calorimeter and the one when it's supposed to exit. The latter one is evaluated with a pure geometrical extrapolation, because we know that the calorimeter material is not added to the Kalman filter. So I think that it makes more sense to use the _pathLengthEntrance and name the function "entranceMomentum(Err)" instead.